internal/driver/fetch_test.go

```diff
@@ -206,6 +206,109 @@ func TestFetch(t *testing.T) {
 	}
 }
 
+func TestFetchWithBase(t *testing.T) {
+	const path = "testdata/"
+	type testcase struct {
+		desc          string
+		sources       []string
+		bases         []string
+		normalize     bool
+		expectSamples bool
+	}
+
+	testcases := []testcase{
+		{
+			"not normalized, base is same as source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu"},
+			false,
+			false,
+		},
+		{
+			"not normalized, single source, multiple base (all profiles same)",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu",path + "cppbench.cpu"},
+			false,
+			true,
+		},
+		{
+			"not normalized, different base and source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "go.crc32.cpu"},
+			false,
+			true,
+		},
+		{
+			"normalized, base is same as source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu"},
+			true,
+			false,
+		},
+		{
+			"normalized, single source, multiple base (all profiles same)",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu",path + "cppbench.cpu"},
+			true,
+			false,
+		},
+		{
+			"normalized, different base and source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "go.crc32.cpu"},
+			true,
+			true,
+		},
+	}
+
+	for _, tc := range testcases {
+		baseVars := pprofVariables
+		pprofVariables = baseVars.makeCopy()
+		defer func() { pprofVariables = baseVars }()
```

Comment:
This usage of defer() here might be a bit misleading. It defers things until the test function exits, not until the for loop iteration finishes. So when the test finishes, there will be a bunch of deferred calls where the last one will be the first call deferred, which will restore the pprof variables to the right, oldest value. But it would probably enough to restore pprofVariables once, with a single defer call at the beginning of the test function?